src/runtime.ts

```diff
@@ -7,41 +7,47 @@ import * as reflection from './reflection-type';
 export { make, server, client, valueClass, reflection };
 
 export const noContentContentType = 'oatsNoContent' as const;
-export function noContent<Status extends number>(
-  status: Status
-): server.Response<Status, typeof noContentContentType, null> {
+export function noContent<Status extends number, Headers extends object>(
+  status: Status,
+  headers?: Headers
+): server.Response<Status, typeof noContentContentType, null, Headers> {
```

Comment:
hmm. isn't this a bit of a lie if there happens to be mandatory headers?

Reply:
I think these might be better typed with providing a no headers signature separately. Like

```
function noContent<Status>(status: Status): Response<Status, nocontenttype, null, null>
``` 

and doing the same for other helpers

Reply:
If there are mandatory headers and we dont pass any, the type will ve infered to {}, which will show an error if you then try to return that in a route

Reply:
if we do specify the type when calling noContent though (noContent<blah,blah,blah>) and then dont pass any headers, then the typechecker wont complain 🙈

Reply:
we could put on some type level tests for this behaviour like we do for client arguments in https://github.com/smartlyio/oats-runtime/blob/master/test/client.spec.ts

also maybe some actual tests to pin down the defaulting behaviour

Reply:
Added some